### ISSUE TITLE
Automagically remove F# compilation resources

### ISSUE THREAD

**Issue by dsyme** [dotnet/linker#329]
The linker can/should automatically remove any .NET manifest assembly resources named `FSharpSignatureData.*` and `FSharpOptimizationData.*` during linking.  

These are only ever needed for F# compilation which doesn't happen on-device and given the aims of the linker it is reasonable to assume this data is unreachable.  

My understanding is that removing these would trim about 1.5MB off the size of all applications using FSharp.Core.dll, and about 3MB off all applications using Elmish.XamarinForms.dll.

Here's an example of how the resources look in ILDASM:
```
    .mresource public FSharpSignatureData.Elmish.XamarinForms
    .mresource public FSharpOptimizationData.Elmish.XamarinForms
```

Thanks

**Comment by nosami** [comment]
I think we need an option to preserve the resources as @fak's Continuous IDE _does_ do F# compilation on device.

That's the only app that I am aware of that might break.

**Comment by dsyme** [comment]
Yes. I'm still amazed that works.  We could recommend that that app bundle a private copy of FSharp.Core but I'm also ok with there being an appropriate option

**Comment by radekdoulik** [comment]
We can extend the custom linker xml definition to be able to preserve resources as well and remove these by default

**Comment by Numpsy** [comment]
Question: Could/Should this linker functionality be extended to strip the new compressed equivalents of these resources (which have different names?)

refs https://github.com/dotnet/fsharp/issues/15605 https://github.com/dotnet/fsharp/issues/17995

**Comment by T-Gro** [comment]
Yes, it should.
The names which can appear in a F# assembly are:

FSharpSignatureData.*
FSharpSignatureDataB.*
FSharpSignatureCompressedData.*
FSharpSignatureCompressedDataB.*

FSharpOptimizationData.*
FSharpOptimizationDataB.*
FSharpOptimizationCompressedData.*
FSharpOptimizationCompressedDataB.*

FSharpOptimizationInfo.*
FSharpSignatureInfo.*
[the last 2 can only ever appear on FSharp.Core and has a different name for historical backwards compat reason, but then also FSharp.Core is what will be present in all compilation units]

**Comment by T-Gro** [comment]
x.StartsWith("FSharpSignature") || x.StartsWith("FSharpOptimization")

Should cover all 10 possibilities.
